learning_resources/views.py

```diff
@@ -179,4 +205,104 @@ def get_queryset(self):
+        """
+        return self._get_base_queryset(
+            resource_type=LearningResourceType.learning_path.value,
+        )
+
+    def list(self, request, *args, **kwargs):
+        """Get all learningpaths for learningpath editors, public learningpaths for all others"""
+
+        if is_learning_path_editor(request) or is_admin_user(request):
+            queryset = self.get_queryset()
+        else:
+            queryset = self.get_queryset().filter(published=True)
+
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
+
+    def create(self, request, *args, **kwargs):
+        request.data["readable_id"] = uuid4().hex
+        request.data["resource_type"] = LearningResourceType.learning_path.value
+        return super().create(request, *args, **kwargs)
+
+    def perform_destroy(self, instance):
+        # deindex_staff_list(instance.learning_resource)
+        instance.delete()
+
+
+class ResourceListItemsViewSet(NestedViewSetMixin, viewsets.ModelViewSet):
+    """
+    Viewset for LearningResource related resources
+    """
+
+    permission_classes = (AnonymousAccessReadonlyPermission,)
+    serializer_class = LearningResourceChildSerializer
+    queryset = (
+        LearningResourceRelationship.objects.select_related("child")
+        .prefetch_related(
+            "child__runs",
+            "child__runs__instructors",
+        )
+        .order_by("position")
+    )
+    pagination_class = DefaultPagination
+
+
+class LearningPathItemsViewSet(ResourceListItemsViewSet):
+    """
+    Viewset for LearningPath related resources
+    """
+
+    serializer_class = LearningResourceRelationshipSerializer
+    permission_classes = (permissions.HasLearningPathItemPermissions,)
+
+    def create(self, request, *args, **kwargs):
+        parent_id = kwargs[f"{DRF_NESTED_PARENT_LOOKUP_PREFIX}parent_id"]
+        request.data["parent"] = parent_id
+        request.data[
+            "relation_type"
+        ] = LearningResourceRelationTypes.LEARNING_PATH_ITEMS.value
+
+        return super().create(request, *args, **kwargs)
+
+    def update(self, request, *args, **kwargs):
+        parent_id = kwargs[f"{DRF_NESTED_PARENT_LOOKUP_PREFIX}parent_id"]
+        request.data["parent"] = parent_id
+        request.data[
+            "relation_type"
+        ] = LearningResourceRelationTypes.LEARNING_PATH_ITEMS.value
+        return super().update(request, *args, **kwargs)
+
+    def perform_destroy(self, instance):
+        instance.delete()
+        # Uncomment when search is ready
+        # learning_path = instance.parent
+        # if learning_path.items.count() > 0:
+        #     upsert_staff_list(staff_list.id)
+        # else:
+        #     deindex_staff_list(staff_list)
```

Comment:
These should really be part of the serializer

Reply:
I think the serializer isn't called when a delete is performed?